### DISCUSSION TITLE
Setting up Docker Content Trust for Grafana Labs on Docker Hub

### DISCUSSION THREAD

**Discussion by Unactived** [grafana/grafana#58529] [General]
Hello,

Many (all?) Grafana Labs products are available and often used as Docker images to easily pull from Docker Hub.

Pulling pre-built images from Docker Hub causes the same supply chain risks than from any other image/package/software source. Some might be concerned about the risk of the supplier platform or account on it being taken over to upload maliciously modified images.

Docker CLI and Docker Hub support Docker Content Trust (DCT : https://docs.docker.com/engine/security/trust/) to easily incorporate verification of integrity and authenticity of images pulled from the Hub. It may also be incorporated in builds and pushes to Docker Hub (https://docs.docker.com/engine/security/trust/trust_automation/).

This doesn't seem to be in place for Grafana and probably Grafana Labs products in general: 
```
# export DOCKER_CONTENT_TRUST=1
# docker pull grafana/grafana:latest
Error: remote trust data does not exist for docker.io/grafana/grafana: notary.docker.io does not have trust data for docker.io/grafana/grafana
```

Has this feature been considered for Grafana Labs and some of their products?

Given how they generally seem to care about security and supply chain issues (e.g. https://grafana.com/tags/security/ or how Grafana plugins support signatures and might enforce it in the future), I hope this is viewed as a good idea.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
